python/cuml/test/test_coordinate_descent.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+import cudf
+import numpy as np
+import pandas as pd
+from cuml import Lasso as cuLasso
+from sklearn.linear_model import Lasso
+from cuml.linear_model import ElasticNet as cuElasticNet
+from sklearn.linear_model import ElasticNet
+from cuml.test.utils import array_equal
+from sklearn.datasets import make_regression
+
+
+@pytest.mark.parametrize('datatype', [np.float32, np.float64])
+@pytest.mark.parametrize('X_type', ['dataframe', 'ndarray'])
+@pytest.mark.parametrize('lr', [0.1, 0.001])
+@pytest.mark.parametrize('algorithm', ['cyclic', 'random'])
+def test_lasso(datatype, X_type, lr, algorithm,
```

Comment:
It would be really cool if we could integrate stress tests directly to the parameterization rather than having to explicitly add a run_stress params and do conditionals on it. I worry that the `if`s will be hard to maintain.

What if we used a "stress" test pytest mark (see the description of "slow" test marks here: https://docs.pytest.org/en/latest/example/simple.html)
then you could parameterize like
```
@pytest.mark.parameterize('nrows', [5000, pytest.param(400000, marks=pytest.mark.stress)
```
(see the "Set marks or test ID for individual parametrized test" discussion   here: https://docs.pytest.org/en/latest/example/parametrize.html)

then we could get rid of all the ifs

Reply:
great idea! I will update the code.